### ISSUE TITLE
A bug in svm.java in sigmoid_train method

### ISSUE THREAD

**Issue by mahan66** [cjlin1/libsvm#37]
If anybody wants to use the probability outputs of the svm, it goes wrong and the output label is always the same. The problem is in the line 1672:

```
if (iter>=max_iter)
	//svm.info("Reaching maximal iterations in two-class probability estimates\n");

probAB[0]=A;probAB[1]=B;
```

As you can see, the next line after the commented part of the if statement will be fired if the statement in the if is true. Thus, the results are always wrong. 

**Solution**
Simply, put comment for the whole if statement.

Regards,
Mahmood

**Comment by kousu** [comment]
!
That's very interesting. I just stumbled across a similar issue with svm.cpp, but it seems to be the reverse problem there: probability predictions are correct, non-probability predictions are always the same class, **but only on certain datasets**. I documented it at https://github.com/scikit-learn/scikit-learn/issues/4800

**Comment by cjlin1** [comment]
There is no problem there. Even if the max num is reached, we still
need to return A and B. The if statement prints just a warning message.

mahan66 writes:

**Comment by kousu** [comment]
I think @mahan66 is pointing out that the comment accidentally changes the meaning. The code should be
```
if (iter>=max_iter) {
    //svm.info("Reaching maximal iterations in two-class probability estimates\n");
}

probAB[0]=A;probAB[1]=B;
```

**Comment by mricherzhagen** [comment]
Yep. there is definitely a problem. The compiler interprets this as 

```
if (iter>=max_iter) {
    //svm.info("Reaching maximal iterations in two-class probability estimates\n");
    probAB[0]=A;
}

probAB[1]=B;
```

**Comment by cjlin1** [comment]
but right now in svm.java the svm.info statement isn't commented out?
Matthias Richerzhagen writes:

**Comment by mahan66** [comment]
Yes, I checked the current source code now and it seems it is correct here. I don't know what version I was using which somebody had made this change.
